[COMGR][debug][test_gpu_reference_kernel] compiler errors

Many errors from compiler during the test_gpu_reference_kernel execution in COMGR build.



ROCm:4.0
gfx906
Runtime:Hip

Not reproduced in CI docker Rocm 3.7.
Reproduced at Rocm 4.2

```
$ CXX=/opt/rocm/llvm/bin/clang++ CXXFLAGS=-Werror cmake -DMIOPEN_TEST_FLAGS=' --disable-verification-cache --verbose' -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS_DEBUG='-g -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=undefined' -DBUILD_DEV=On -DMIOPEN_GPU_SYNC=Off -DMIOPEN_USE_COMGR=On ../MLOpen
$ ./bin/test_gpu_reference_kernel
```

```
....
MIOpen(HIP): Info [ConvolutionWrwImmediate] solver_id = ConvDirectNaiveConvWrw, workspace = 0
MIOpen(HIP): Info [FindSolutionImpl] ConvDirectNaiveConvWrw (not searchable)
n:2, c:8, di:8, hi:14, wi:14, k:3, do:4, ho:7, wo:6, fz:3, fy:3,fx:3, pz:1, py:1, px:1, sz:2, sy:2, sx:2, dz:1, dy:1, dx:2, g:1, dir:wrw, type:bf16, layout:NCDHW, valid:1
MIOpen(HIP): Info [get_device_name] Raw device name: gfx906
MIOpen(HIP): Info [Handle] stream: 0x737e0d0, device_id: 0
MIOpen(HIP): Info [ConvolutionForwardImmediate] solver_id = ConvDirectNaiveConvFwd, workspace = 0
MIOpen(HIP): Info [FindSolutionImpl] ConvDirectNaiveConvFwd (not searchable)
MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_TO_BC: ERROR (1)
MIOpen(HIP): Error [BuildHip] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildHip] In file included from /tmp/comgr-221397/input/naive_conv.cpp:26:
In file included from /opt/rocm/hip/include/hip/hip_fp16.h:29:
In file included from /opt/rocm/hip/include/hip/hcc_detail/hip_fp16.h:32:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/algorithm:60:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/utility:69:
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:87:7: error: redefinition of 'operator!='
      operator!=(const _Tp& __x, const _Tp& __y)
      ^
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:87:7: note: previous definition is here
      operator!=(const _Tp& __x, const _Tp& __y)
      ^
In file included from /tmp/comgr-221397/input/naive_conv.cpp:26:
In file included from /opt/rocm/hip/include/hip/hip_fp16.h:29:
In file included from /opt/rocm/hip/include/hip/hcc_detail/hip_fp16.h:32:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/algorithm:60:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/utility:69:
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:100:7: error: redefinition of 'operator>'
      operator>(const _Tp& __x, const _Tp& __y)
      ^
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:100:7: note: previous definition is here
      operator>(const _Tp& __x, const _Tp& __y)
      ^
In file included from /tmp/comgr-221397/input/naive_conv.cpp:26:
In file included from /opt/rocm/hip/include/hip/hip_fp16.h:29:
In file included from /opt/rocm/hip/include/hip/hcc_detail/hip_fp16.h:32:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/algorithm:60:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/utility:69:
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:113:7: error: redefinition of 'operator<='
      operator<=(const _Tp& __x, const _Tp& __y)
      ^
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:113:7: note: previous definition is here
      operator<=(const _Tp& __x, const _Tp& __y)
      ^
In file included from /tmp/comgr-221397/input/naive_conv.cpp:26:
In file included from /opt/rocm/hip/include/hip/hip_fp16.h:29:
In file included from /opt/rocm/hip/include/hip/hcc_detail/hip_fp16.h:32:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/algorithm:60:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/utility:69:
/usr/lib/gcc/x86_64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/stl_relops.h:126:7: error: redefinition of 'operator>='
      operator>=(const _Tp& __x, const _Tp& __y)
...
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -45,7 +45,6 @@ option( MIOPEN_TEST_DRIVER_ITER_MODE Off)
 option( MIOPEN_TEST_MIOTENSILE "Test MIOpenTensile path" OFF )
 option( MIOPEN_TEST_MLIR "Test for MLIR compilation backend" ${MIOPEN_USE_MLIR} )
 
-option( WORKAROUND_ISSUE_898 "" ON)
 option( WORKAROUND_ISSUE_936 "" ON)
 option( WORKAROUND_ISSUE_1053 "" ON)
 
@@ -271,10 +270,6 @@ function(add_test_command NAME EXE)
         else()
             add_test(NAME ${NAME} COMMAND ${EXE} ${ARGN})
         endif()
-    endif()
-
-    if(WORKAROUND_ISSUE_898 AND MIOPEN_USE_COMGR)
-        set_property(TEST ${TEST_NAME} PROPERTY ENVIRONMENT MIOPEN_DEBUG_COMGR_HIP_PCH_ENFORCE=0)
     endif()
 endfunction()
 
@@ -566,10 +561,6 @@ function(add_custom_test NAME)
 
     add_custom_target(${NAME} ${PARSE_UNPARSED_ARGUMENTS})
     add_test(NAME ${NAME} COMMAND ${CMAKE_COMMAND} --build ${CMAKE_CURRENT_BINARY_DIR} --target ${NAME})
-
-    if(WORKAROUND_ISSUE_898 AND MIOPEN_USE_COMGR)
-        set_property(TEST ${NAME} PROPERTY ENVIRONMENT MIOPEN_DEBUG_COMGR_HIP_PCH_ENFORCE=0)
-    endif()
 
     if(WORKAROUND_ISSUE_1148
         AND (${NAME} MATCHES "test_conv_3d"

